Framework/Core/src/DataProcessor.cxx

```diff
@@ -39,7 +38,7 @@ void DataProcessor::doSend(FairMQDevice &device, RootObjectContext &context) {
     assert(message.payload.get());
     FairMQParts parts;
     FairMQMessagePtr payload(device.NewMessage());
-    TClonesArray *a = reinterpret_cast<TClonesArray*>(message.payload.get());
+    auto a = reinterpret_cast<TObject*>(message.payload.get());
```

Comment:
payload is of type `unique_ptr<TObject>`, so no cast needed

Reply:
isn't payload a (void*)?